src/language/enums.md

```diff
@@ -0,0 +1,69 @@
+# Struct and enums
+
+hacspec also supports user-defined structs and enums with some restrictions.
+
+## Structs
+
+The only form of struct declaration currently allowed in hacspec is:
+
+```rust, noplaypen
+struct Foo(u32, Seq<u32>);
+```
+
+The struct thus declared can have one ore more components. This form of struct
+declaration effectively corresponds to a single-case enum, and is implemented
+as such. Struct components can be accessed through let-binding destructuring:
+
+```rust, noplaypen
+let Foo(x, y) = z;
+```
+
+Note that you can't store borrowed types inside hacspec structs, hence there is no
+need for lifetime variables.
+
+## Enums
+
+hacspec supports the basic `enum` declarations of Rust:
```

Comment:
Not sure if basic enums is specific enough. C-style with assignments for example might be considered basic but aren't allowed.